src/game/game_objects.rs

```diff
@@ -0,0 +1,75 @@
+use bevy::prelude::{Component, Entity};
+
+#[derive(Eq, Hash)]
+pub enum GameObjects {
+    Box(Option<Entity>),
+    Wall,
+    Empty,
+}
+
+#[derive(Component)]
+pub struct Goal;
+
+impl PartialEq for GameObjects {
+    fn eq(&self, other: &Self) -> bool {
+        match (self, other) {
+            (Self::Box(_), Self::Box(_)) => true, //we do not care about the insides of Box when comapring
+            (Self::Wall, Self::Wall) => true,
+            (Self::Empty, Self::Empty) => true,
+            _ => false,
+        }
+    }
+}
```

Comment:
skoro nadpisujecie PartialEq to warto się zastanowić, czy ten enum jest dobrym miejscem do trzymania tych danych, które tutaj ignorujecie, in the first place